### PR TITLE
 Modify S2078(Common text): Update to the education framework (APPSEC-121)

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**

### CODE REVIEW

File: rules/S2078/common/fix/validation.adoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+* `(`
+* `)`
+* `\\`
+* `null`
+
+For Java, OWASP's functions
+https://www.javadoc.io/doc/org.owasp.esapi/esapi/latest/org/owasp/esapi/Encoder.html#encodeForDN-java.lang.String-[`encodeForDN`]
+and
+https://www.javadoc.io/doc/org.owasp.esapi/esapi/latest/org/owasp/esapi/Encoder.html#encodeForLDAP-java.lang.String-[`encodeForLDAP`]
+allow sanitizing these characters automatically.
```
Comment (loris-s-sonarsource) [inline rules/S2078/common/fix/validation.adoc:52]: Needs discussion

File: rules/S2078/impact.adoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
+=== What are the possible effects?
```
Comment (loris-s-sonarsource) [inline rules/S2078/impact.adoc:1]:
```suggestion
=== What is the possible impact?
```

**Review by pedro-oliveira-sonarsource — CHANGES_REQUESTED** [review]
Just some suggestions for you to review.

File: rules/S2078/common/fix/validation.adoc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+* `=`
+
+For **Search Filters**, special characters that need to be escaped include:
+
+* `\``
+* `(`
+* `)`
+* `\\`
+* `null`
+
```
[the remark this answers is not in the document]
Comment (pedro-oliveira-sonarsource) [reply rules/S2078/common/fix/validation.adoc:47]: Disregard this, saw you already added that on the Java how-to-fix-it section which comes after including the validation.adoc.

[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S2078/common/fix/validation.adoc:47]: Ok, and for dotnet we don't actually support one yet :(

**Review by pedro-oliveira-sonarsource — APPROVED** [review]
LGTM.

**Pull request merged**